sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/tests/Bindings/ServiceBusTriggerAttributeBindingProviderTests.cs

```diff
@@ -108,6 +108,43 @@ public async Task GetAutoCompleteMessagesOptionToUse_AutoCompleteDisabledOnTrigg
             Assert.False(autoCompleteMessagesFlagSentToListener);
         }
 
+        [Test]
+        public async Task GetMaxMessageBatchSizeOptionToUse_MaxMessageBatchSizeOnTrigger()
+        {
+            var listenerContext = new ListenerFactoryContext(
+                new Mock<FunctionDescriptor>().Object,
+                new Mock<ITriggeredFunctionExecutor>().Object,
+                CancellationToken.None);
+            var parameters = new object[] { listenerContext };
+            var entityPath = "maxMessageBatchSizeOnTrigger";
+
+            var client = new ServiceBusClient(_testConnection);
+            ServiceBusProcessor processor = client.CreateProcessor(entityPath);
+            ServiceBusReceiver receiver = client.CreateReceiver(entityPath);
+            var configuration = ConfigurationUtilities.CreateConfiguration(new KeyValuePair<string, string>("connection", _testConnection));
+
+            ServiceBusOptions config = new ServiceBusOptions();
+            var mockMessageProcessor = new Mock<MessageProcessor>(MockBehavior.Strict, processor);
+
+            var mockMessagingProvider = new Mock<MessagingProvider>(new OptionsWrapper<ServiceBusOptions>(config));
+            var mockClientFactory = new Mock<ServiceBusClientFactory>(configuration, Mock.Of<AzureComponentFactory>(), mockMessagingProvider.Object, new AzureEventSourceLogForwarder(new NullLoggerFactory()), new OptionsWrapper<ServiceBusOptions>(new ServiceBusOptions()));
+            mockMessagingProvider
+                .Setup(p => p.CreateMessageProcessor(It.IsAny<ServiceBusClient>(), entityPath, It.IsAny<ServiceBusProcessorOptions>()))
+                .Returns(mockMessageProcessor.Object);
+
+            var parameter = GetType().GetMethod("TestMaxMessageBatchSizeOnTrigger", BindingFlags.NonPublic | BindingFlags.Static).GetParameters()[0];
+            var context = new TriggerBindingProviderContext(parameter, CancellationToken.None);
+            var binding = await _provider.TryCreateAsync(context);
+            var createListenerTask = binding.GetType().GetMethod("CreateListenerAsync");
+            var listener = await (Task<IListener>)createListenerTask.Invoke(binding, parameters);
+            var listenerOptions = (ServiceBusOptions)listener.GetType().GetField("_serviceBusOptions", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(listener);
+            var maxMessageBatchSizeSentToListener = (int)listener.GetType().GetField("_maxMessageBatchSize", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(listener);
+
+            Assert.NotNull(listenerOptions);
+            Assert.AreEqual(listenerOptions.MaxMessageBatchSize, 1000);
+            Assert.AreEqual(maxMessageBatchSizeSentToListener, 2);
```

Comment:
Do we also need a negative test that verifies when the trigger is not set, the host.json option is used?